packages/amplify_datastore/ios/Classes/types/model/FlutterModelField.swift

```diff
@@ -20,65 +20,88 @@ import Amplify
 public struct FlutterModelField {
     
     public let name: String
-    public let type: ModelFieldType
+    public let type: FlutterModelFieldType
     public let isRequired: Bool
     public let isArray: Bool
+    public let association: FlutterModelAssociation?
+    public let authRules: [FlutterAuthRule]?
+    
+    // Currently ignored
     public let attributes: [ModelFieldAttribute] = []
-    public let association: ModelAssociation? = nil
-    public let authRules: AuthRules = []
+
     
     init(serializedData: [String: Any]) {
         self.name = serializedData["name"] as! String
-        self.type = ModelFieldType.from(type: FlutterModelField.deserializeFieldType(type: serializedData["type"] as! String))
+        
+        self.type = FlutterModelFieldType(serializedData: serializedData["type"] as! [String: Any])
+        
         self.isRequired = serializedData["isRequired"] as! Bool
         self.isArray = serializedData["isArray"] as! Bool
+        
+        self.association = serializedData["association"] != nil ?  FlutterModelAssociation(serializedData: serializedData["association"] as! [String : Any]) :
+            nil
```

Comment:
avoid force casts throughout.

Reply:
what would you recommend as replacement?  It appears guard does not work in this scenario